### ISSUE TITLE
AsyncEventFilterBuilder.deploy raises Invalid web3 argument

### ISSUE THREAD

**Issue by stibala** [ethereum/web3.py#2944]
* Version: 6.0.0.
* Python: 3.10.9
* OS: osx
* `pip freeze` output

```
aiohttp==3.8.4
aiosignal==1.3.1
async-timeout==4.0.2
attrs==23.1.0
bitarray==2.7.3
certifi==2022.12.7
charset-normalizer==3.1.0
cytoolz==0.12.1
eth-abi==4.0.0
eth-account==0.8.0
eth-hash==0.5.1
eth-keyfile==0.6.1
eth-keys==0.4.0
eth-rlp==0.3.0
eth-typing==3.3.0
eth-utils==2.1.0
frozenlist==1.3.3
hexbytes==0.3.0
idna==3.4
importlib-resources==5.12.0
jsonschema==4.17.3
lru-dict==1.1.8
multidict==6.0.4
parsimonious==0.9.0
protobuf==4.22.3
pycryptodome==3.17
pyrsistent==0.19.3
regex==2023.5.4
requests==2.29.0
rlp==3.0.0
toolz==0.12.0
urllib3==1.26.15
web3==6.2.0
websockets==11.0.2
yarl==1.9.2
```

### What was wrong?

```python
import asyncio
from web3 import AsyncWeb3, AsyncHTTPProvider, Web3

RPC_ADDRESS = "<rpc node address>"
ADDRESS = "<contract address>"
ABI =  [{...}]

async def example():
    web3 = AsyncWeb3(AsyncHTTPProvider(RPC_ADDRESS))
    contract = web3.eth.contract(address=ADDRESS, abi=ABI)
    # this causes the error in web3/_utils/events.py at line 451
    my_filter = await contract.events.Transfer.create_filter(fromBlock=0)
    entries = await my_filter.get_new_entries()
    print(entries)

asyncio.run(example())
```

raises `ValueError: Invalid web3 argument: got: <web3.main.AsyncWeb3 object at ...`

### How can it be fixed?

The check in web3/_utils/events.py at line 450 `if not isinstance(async_w3, web3.Web3):` is wrong in my opinion. If changed to `if not isinstance(async_w3, web3.AsyncWeb3):` the code works.

**Comment by fselmo** [comment]
This has been reported (#2931) and fixed (#2934), we should have a release out with it soon. Thanks for independently reporting though 👍🏼

**Comment by fselmo** [comment]
@stibala, it is now released in version `6.3.0` 🙂

**Comment by stibala** [comment]
thanks guys! Didn't see that issue.
